File: src/main/java/io/github/takusan23/clickmanaita/ClickManaita.java
package io.github.takusan23.clickmanaita;

import io.github.takusan23.clickmanaita.block.ClickManaitaBlocks;
import io.github.takusan23.clickmanaita.creativetab.ClickManaitaCreativeTab;
import io.github.takusan23.clickmanaita.enchant.ClickManaitaEnchant;
import io.github.takusan23.clickmanaita.item.ClickManaitaItems;
import net.minecraftforge.common.MinecraftForge;
import net.minecraftforge.eventbus.api.bus.BusGroup;
import net.minecraftforge.fml.common.Mod;
import net.minecraftforge.fml.javafmlmod.FMLJavaModLoadingContext;

@Mod(ClickManaita.MOD_ID)
public class ClickManaita {

    /**
     * MODのID
     */
    public static final String MOD_ID = "clickmanaita";

    /**
     * コンストラクタ
     */
    public ClickManaita(FMLJavaModLoadingContext context) {
        BusGroup modBusGroup = context.getModBusGroup();
        // ブロック登録
        ClickManaitaBlocks.register(modBusGroup);
        // アイテム（ブロックのアイテム）登録
        ClickManaitaItems.register(modBusGroup);
        // クリエイティブタブ登録
        ClickManaitaCreativeTab.register(modBusGroup);
        // エンチャント（カスタムエフェクト、エフェクトコンポーネント）を登録
        ClickManaitaEnchant.register(modBusGroup);
        // プレイヤーイベント
        MinecraftForge.EVENT_BUS.register(new ClickManaitaPlayerEvent());
    }
}

File: src/main/kotlin/io/github/takusan23/clickmanaita/ClickManaita.kt
package io.github.takusan23.clickmanaita

import io.github.takusan23.clickmanaita.block.ClickManaitaBlock
import io.github.takusan23.clickmanaita.enchant.ClickManaitaEnchantClickCallback
import io.github.takusan23.clickmanaita.enchant.ClickManaitaEnchantEntityEffect
import io.github.takusan23.clickmanaita.enchant.EnchantRightClickEffectComponent
import io.github.takusan23.clickmanaita.item.ClickManaitaItem
import net.minecraft.core.registries.BuiltInRegistries
import net.minecraft.core.Registry
import net.minecraft.resources.Identifier

/**
 * エントリーポイント。起動時にinit関数が呼ばれる
 */
@Suppress("unused")
fun init() {
    // アイテム追加
    ClickManaitaItem.register()
    // ブロック追加
    ClickManaitaBlock.register()
    // クリエタブ
    Registry.register(BuiltInRegistries.CREATIVE_MODE_TAB, Identifier.fromNamespaceAndPath("clickmanaita", "clickmanaita_creative_tab"), ClickManaitaItemGroup.CLICKMANAITA_ITEMGROUP)
    // エンチャントのカスタムエフェクト、カスタムエフェクトのトリガー条件を追加
    Registry.register(BuiltInRegistries.ENCHANTMENT_ENTITY_EFFECT_TYPE, Identifier.fromNamespaceAndPath("clickmanaita", "clickmanaita_enchant_effect"), ClickManaitaEnchantEntityEffect.CODEC)
    Registry.register(BuiltInRegistries.ENCHANTMENT_EFFECT_COMPONENT_TYPE, Identifier.fromNamespaceAndPath("clickmanaita", "block_right_click"), EnchantRightClickEffectComponent.BLOCK_RIGHT_CLICK_EFFECT_COMPONENT)
    // イベント登録
    ClickManaitaEnchantClickCallback.registerClickManaitaEnchantCallback()
}

